### PR TITLE
Clear successful / failed uploads bug fix

### PR DESCRIPTION
**Pull request by sd1998**
Fix to issue #2563.

### CODE REVIEW

**Comment by sd1998** [comment]
@davigonz @jesmrec Fix for #2563.

**Comment by jesmrec** [comment]
Thanks a lot!! we planned to include this in 2.11, so this is amazing!. Please, stay tunned, just in case we need some change in the code review and QA. We are interested in merging this asap. Thanks again!

**Comment by jesmrec** [comment]
@sd1998 you are member of the repo, so you can send your PRs from a clone instead from a fork. You decide how to do it, but for the team could be helpful in order to rebase the branches after the reviews without bothering you every time your work is outdated with `master`. Up2you

**Comment by davigonz** [comment]
Thanks @sd1998 for this contribution, the code looks good to me so far, but could you fix https://github.com/owncloud/android/issues/2568 also here since the solution would be pretty similar?

**Comment by sd1998** [comment]
@davigonz I have added the fix for #2568.

File: owncloudApp/src/main/java/com/owncloud/android/datamodel/UploadsStorageManager.java [hunk 1]
```diff
@@ -428,9 +428,10 @@ private ContentResolver getDB() {
     public long clearFailedButNotDelayedForWifiUploads() {
         long result = getDB().delete(
                 ProviderTableMeta.CONTENT_URI_UPLOADS,
-                ProviderTableMeta.UPLOADS_STATUS + "==" + UploadStatus.UPLOAD_FAILED.value + " AND " +
-                        ProviderTableMeta.UPLOADS_LAST_RESULT + "<>" + UploadResult.DELAYED_FOR_WIFI.getValue(),
-                null
+                ProviderTableMeta.UPLOADS_STATUS + "= ? AND " +
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/datamodel/UploadsStorageManager.java]: Could you delete the space between `=` and `?` and `!=` and `?`

**Comment by sd1998** [comment]
@davigonz I have made the changes.

**Comment by davigonz** [comment]
Thanks @sd1998 for the changes, @jesmrec ready to test

**Comment by jesmrec** [comment]
Crash does not happen anymore. Approved

**Pull request merged**